### ISSUE TITLE
New `data_info` standard

### ISSUE THREAD

**Issue by fededagos** [open-retina/open-retina#153]
Following discussions at the workshop, we need to implement a new `data_info` standard. Here is a proposal based on previous discussions. @thomasZen and @BaptisteLorenzi: please chime in if you have further input, I would start implementing the necessary changes to save these from next week.

First, I would propose to make the information also available under the alias `pretrained_cfg`, i.e. following similar conventions as [torch image models (`timm`)](https://huggingface.co/docs/timm/en/quickstart).

Then, I would propose saving the following fields in the dictionary:
- `n_neurons` : Dictionary, with sessions names as keys, and number of neurons in the sessions as values. This is already present.
- `mean`: tuple, with mean normalisation values for the videos. Would replace "movie_norm_dict" and "norm_mean" inside that, for ease of access.
- `std`: tuple, same as above. The idea is to flatten a bit the entries and make them more accessible
- `input_shape`: already present. Shape of the video input, excluding time dimension
- `response_rate`: float. Sampling rate of responses (i.e. the one used to train, so does not need to equate original experimental sampling rate for example after upsampling or downsampling from binning).
- `stimulus_rate`: float . Frame rate of the stimulus.
- `stimulus_class`: str, or list of str. The type of stimuli used to train the model, e.g. white noise, natural movies,... I do not see a need to standardise this right now, but we could settle on literal entries to use here. 
- `model_cut_frames`: int. Stimulus frames cut by the model (e.g. as a consequence of convolutions).
- `conv_type`: optional. Entry would be a string, indicating the convolution type the model is using.
- `core_type`: optional. Entry would be a string, indicating the type of core the model is using. 
- `readout_type`: optional. Entry would be a string, indicating which readout class is being used.
- `sessions_info`: optional. Nested dictionary with all extra session information that can be useful for downstream analyses. Keys would be session names, and values other dictionaries containing information either globally for the session (e.g. like the eye), or arrays containing information per neuron. This would be a potential candidate for standardisation.

I would vote for keeping the explainable variance per neuron / model performance of a neuron optional, and potentially going into sessions_info. I think it would still be useful to have an utility function that automatically adds at least the model performance at the end of training. I can add this in the same or a separate PR.

**Comment by thomasZen** [comment]
> conv_type: optional. Entry would be a string, indicating the convolution type the model is using.
core_type: optional. Entry would be a string, indicating the type of core the model is using.
readout_type: optional. Entry would be a string, indicating which readout class is being used.

Isn't it easy to get this from a trained model using `type(model.core)` and `(type(model.readout))`. If so I would not put it in the data_info.

**Comment by thomasZen** [comment]
What would additionally be useful:
- size of one pixel (in reality and on retinal surface, in retinal degrees)
- Maybe: performance metrics for the model on each neuron (to decide whether to "trust" it, also I'm unsure if this is a satisfying solution)

**Comment by BaptisteLorenzi** [comment]
The size of one pixel would be necessary to test new stimulus correctly, so I support this one.
mean/std could have clearer names like stim_mean/stim_std or input_mean/input_std ...
I also don't see why the conv/core types should be part of the data_info.

About the performance metrics, an easier access to the test function would be enough it's true. But it's also very lightweight to save so I don't see why not.

**Comment by fededagos** [comment]
Thanks for the feedback!

I think it makes sense to explicitly state `stim_mean` and `stim_std`, as one could also decide to normalise individual neuron responses for training, and then having `response_mean` and `response_std` would also be possible.

My proposal would then to have three additional fields:
- `retina_pixel_size_um`: required. Projected size of a pixel onto the retina in micrometers.
- `pretrained_peformance`: optional. Entry would be a dictionary with session names as keys, and then one entry per neuron with the performance saved in there
- `pretrained_performance_metric`: optional, string. Name of the metric used to calculate the performance.

Any additional thoughts on this?
